### PR TITLE
feat(CommunityOwnership): Finalise ownership UI flows

### LINKED ISSUE #12174: [Community Ownership] Finalise ownership flows
### Description
- Create `Finalize ownership` popup with all the needed flows (accept transaction / decline to be the owner).
- Itegrate new popup from the [3 different methods](https://www.figma.com/file/qHfFm7C9LwtXpfdbxssCK3/Kuba%E2%8E%9CDesktop---Communities?type=design&node-id=37206%3A87273&mode=design&t=WK5aiGMX4bnXgd41-1) from where the finalisation can happen:

1. From community colum / button
Screenshot 2023-09-15 at 13 07 33

2. From Community Settings `Finalise ownership`
Screenshot 2023-09-15 at 13 07 55

3. From notification center
Screenshot 2023-09-15 at 13 08 35

### PR DESCRIPTION
**Pull request by noeliaSD**
Closes #12174 

### What does the PR do
- Created finalise popup and added it into storybook.
- Decline popup created and added support in storybook.
- Itegration of the finalise flow in the app from: community column button and from community overview footer button.
- Updated storybook pages (settings overview related) according to new components requirements.

**NOTE: Only UI, backend not ready. Navigation from activity center not covered (missing backend).**

### Affected areas

Community Column / Community Overview

### Screenshot of functionality 

- App:

https://github.com/status-im/status-desktop/assets/97019400/56053dec-3d61-4d67-afff-7558f8d7d94c

- Storybook:

https://github.com/status-im/status-desktop/assets/97019400/ab124bc0-2709-483a-ab1e-1fdae4b3c6cc

https://github.com/status-im/status-desktop/assets/97019400/ac1b397d-e3d2-4355-a09c-a513a8263bd3

### CODE REVIEW

File: ui/app/AppLayouts/Communities/popups/FinaliseOwnershipDeclinePopup.qml [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+
+import utils 1.0
+
+StatusDialog {
+    id: root
+
+    // Community related props:
+    required property string communityName
+
+    signal cancelClicked
```
Comment (caybro) [inline ui/app/AppLayouts/Communities/popups/FinaliseOwnershipDeclinePopup.qml:18]: Need to handle that signal? It just closes the popup w/o doing anything right

File: ui/app/AppLayouts/Communities/popups/FinaliseOwnershipDeclinePopup.qml [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+
+    width: 480 // by design
+    padding: Style.current.padding
+    contentItem: StatusBaseText {
+        wrapMode: Text.WrapAtWordBoundaryOrAnywhere
+        text: qsTr("If you don’t want to be the owner of the %1 Community it is important that you let the previous owner know so they can organise another owner to take over. You will have to send the Owner token back to them or on to the next designated owner.").arg(root.communityName)
+        lineHeight: 1.2
+    }
+
+    header: StatusDialogHeader {
```
Comment (caybro) [inline ui/app/AppLayouts/Communities/popups/FinaliseOwnershipDeclinePopup.qml:29]: You don't need a custom header here, just set `title` in `root`

File: ui/app/AppLayouts/Communities/popups/FinaliseOwnershipPopup.qml [hunk 3]
```diff
@@ -0,0 +1,335 @@
...
+
+    signal finaliseOwnershipClicked
+    signal rejectClicked
+    signal visitCommunityClicked
+    signal openControlNodeDocClicked(string link)
+
+    QtObject {
+        id: d
+
+        readonly property string controlNodeLink: "https://help.status.im/en/status-communities/about-the-control-node-in-status-communities/"
```
Comment (caybro) [inline ui/app/AppLayouts/Communities/popups/FinaliseOwnershipPopup.qml:46]: Are you sure the help URL is correct? We already have a different one in `Constants.statusHelpLinkPrefix`

File: ui/app/AppLayouts/Communities/popups/FinaliseOwnershipPopup.qml [hunk 4]
```diff
@@ -0,0 +1,335 @@
...
+                onClicked: {
+                    root.rejectClicked()
+                    close()
+                }
+            }
+
+            StatusButton {
+                id: acceptBtn
+
+                type: StatusBaseButton.Type.Normal
```
Comment (caybro) [inline ui/app/AppLayouts/Communities/popups/FinaliseOwnershipPopup.qml:153]: default

File: ui/app/AppLayouts/Communities/popups/FinaliseOwnershipPopup.qml [hunk 5]
```diff
@@ -0,0 +1,335 @@
...
+                    d.accountName = item.name
+                }
+            }
+
+            StatusCheckBox {
+                Layout.topMargin: -Style.current.halfPadding
+                Layout.fillWidth: true
+
+                checked: d.ackCheck
+                font.pixelSize: Style.current.primaryTextFontSize
```
Comment (caybro) [inline ui/app/AppLayouts/Communities/popups/FinaliseOwnershipPopup.qml:328]: Hehe, I fixed that in https://github.com/status-im/status-desktop/pull/12252/commits/ad4f20578950c4cda5c566311f37fbf3bcc9391d

Needs review :)

**Review by caybro — COMMENTED** [review]
Nice work, just some minor comments :)

File: ui/app/AppLayouts/Communities/views/CommunityColumnView.qml [hunk 6]
```diff
@@ -510,22 +445,126 @@ Item {
...
+                    requirementsCheckPending: root.store.requirementsCheckPending
+                    name: communityData.name
+                    introMessage: communityData.introMessage
+                    imageSrc: communityData.image
+                    accessType: communityData.access
+                    loginType: root.store.loginType
+                    walletAccountsModel: WalletStore.RootStore.nonWatchAccounts
+                    permissionsModel: {
+                        root.store.prepareTokenModelForCommunity(communityData.id)
+                        return root.store.permissionsModel
```
Comment (alexjba) [inline ui/app/AppLayouts/Communities/views/CommunityColumnView.qml:521]: I'm wondering why is this a 2 steps action. IMO the `prepare` step is an unnecessary detail for the visual components. Might be a good idea to replace it with a store function `getCommunityPermissionsModel(communityId)` and do the prepare inside (not necessarily in this PR). WDYT?

Comment (caybro) [reply ui/app/AppLayouts/Communities/views/CommunityColumnView.qml:521]: Maybe because the operation itself is async?

Comment (alexjba) [reply ui/app/AppLayouts/Communities/views/CommunityColumnView.qml:521]: It's async and looks to me like a lazy load implementation. But I think it's better to isolate this inside the store.

File: ui/app/AppLayouts/Communities/popups/FinaliseOwnershipPopup.qml [hunk 7]
```diff
@@ -0,0 +1,335 @@
...
+
+                onClicked: {
+                    root.rejectClicked()
+                    close()
+                }
+            }
+
+            StatusButton {
+                id: acceptBtn
+
```
Comment (micieslak) [inline ui/app/AppLayouts/Communities/popups/FinaliseOwnershipPopup.qml:152]: I would add `Layout.fillWidth: implicitWidth > parent.width` here to elide properly when translation is wider than available space.

File: ui/app/AppLayouts/Communities/views/CommunityColumnView.qml (hunk 6 above)
Comment (micieslak) [reply ui/app/AppLayouts/Communities/views/CommunityColumnView.qml:521]: maybe hiding it in `getCommunityPermissionsModel` is a bit better but imo both options looks like a design problem. Ideally it should be enough to just access `permissionsModel` property without taking care about any implementation details in stores/backends.

**Review by micieslak — APPROVED** [review]
Great job, looks good to me, only points already mentioned by @alexjba and @caybro

File: ui/app/AppLayouts/Communities/popups/FinaliseOwnershipDeclinePopup.qml (hunk 1 above)
Comment (noeliaSD) [reply ui/app/AppLayouts/Communities/popups/FinaliseOwnershipDeclinePopup.qml:18]: Removed!

File: ui/app/AppLayouts/Communities/popups/FinaliseOwnershipDeclinePopup.qml (hunk 2 above)
Comment (noeliaSD) [reply ui/app/AppLayouts/Communities/popups/FinaliseOwnershipDeclinePopup.qml:29]: True!

File: ui/app/AppLayouts/Communities/popups/FinaliseOwnershipPopup.qml (hunk 3 above)
Comment (noeliaSD) [reply ui/app/AppLayouts/Communities/popups/FinaliseOwnershipPopup.qml:46]: Yeah! Updated to: `https://status.app/help/status-communities/about-the-control-node-in-status-communities`

File: ui/app/AppLayouts/Communities/views/CommunityColumnView.qml (hunk 6 above)
Comment (noeliaSD) [reply ui/app/AppLayouts/Communities/views/CommunityColumnView.qml:521]: Good point 🤔! I don't have the details for that, just changed it's parent to be a component. I'll create a new issue to check it out in a separte task.

**Comment by noeliaSD** [comment]
Updated according to comments!

**Pull request merged**